src/plugins/unified_histogram/public/chart/breakdown_field_selector.tsx

```diff
@@ -69,16 +85,24 @@ export const BreakdownFieldSelector = ({
     });
 
     return options;
-  }, [dataView, breakdown.field]);
+  }, [fields, breakdown?.field]);
 
   const onChange = useCallback<NonNullable<ToolbarSelectorProps['onChange']>>(
     (chosenOption) => {
-      const field = chosenOption?.value
-        ? dataView.fields.find((currentField) => currentField.name === chosenOption.value)
-        : undefined;
-      onBreakdownFieldChange?.(field);
+      let breakdownField: DataViewField | undefined;
+      if (esqlColumns) {
+        const breakdownColumn = esqlColumns?.find((column) => column.name === chosenOption?.value);
+        breakdownField = breakdownColumn
+          ? new DataViewField(convertDatatableColumnToDataViewFieldSpec(breakdownColumn))
+          : undefined;
+      } else {
+        breakdownField = chosenOption?.value
+          ? dataView.fields.find((currentField) => currentField.name === chosenOption.value)
+          : undefined;
+      }
```

Comment:
We could simplify this: remove `if` case and go through `fields` array instead (considering it consists of data view fields or derived fields with `new DataViewField(convertDatatableColumnToDataViewFieldSpec(column))`)

Reply:
Sure! Done here https://github.com/elastic/kibana/pull/193820/commits/9e731dfc361a48659f74ee8b3896cdc0184cbf79